lib/ansible/modules/cloud/azure/azure_rm_storageaccount.py

```diff
@@ -151,25 +213,57 @@
     # This is handled in azure_rm_common
     pass
 
+import copy
 from ansible.module_utils.azure_rm_common import AZURE_SUCCESS_STATE, AzureRMModuleBase, HAS_AZURE
+from ansible.module_utils._text import to_native
+
+cors_rule_spec = dict(
+    allowed_origins=dict(type='list', elements='str', required=True),
+    allowed_methods=dict(type='list', elements='str', required=True),
+    max_age_in_seconds=dict(type='int', required=True),
+    exposed_headers=dict(type='list', elements='str', required=True),
+    allowed_headers=dict(type='list', elements='str', required=True),
+)
+
+
+def compare_cors(cors1, cors2):
+    if len(cors1) != len(cors2):
+        return False
+    copy2 = copy.copy(cors2)
+    for rule1 in cors1:
+        matched = False
+        for rule2 in copy2:
+            if (rule1['max_age_in_seconds'] == rule2['max_age_in_seconds']
+                    and set(rule1['allowed_methods']) == set(rule2['allowed_methods'])
+                    and set(rule1['allowed_origins']) == set(rule2['allowed_origins'])
+                    and set(rule1['allowed_headers']) == set(rule2['allowed_headers'])
+                    and set(rule1['exposed_headers']) == set(rule2['exposed_headers'])):
+                matched = True
+                copy2.remove(rule2)
+        if not matched:
+            return False
+    return True
 
 
 class AzureRMStorageAccount(AzureRMModuleBase):
 
     def __init__(self):
 
         self.module_arg_spec = dict(
-            account_type=dict(type='str', choices=['Premium_LRS', 'Standard_GRS', 'Standard_LRS', 'StandardSSD_LRS', 'Standard_RAGRS', 'Standard_ZRS'],
+            account_type=dict(type='str',
+                              choices=['Premium_LRS', 'Standard_GRS', 'Standard_LRS', 'StandardSSD_LRS', 'Standard_RAGRS', 'Standard_ZRS', 'Premium_ZRS'],
                               aliases=['type']),
-            custom_domain=dict(type='dict'),
+            custom_domain=dict(type='dict', aliases=['custom_dns_domain_suffix']),
             location=dict(type='str'),
             name=dict(type='str', required=True),
             resource_group=dict(required=True, type='str', aliases=['resource_group_name']),
             state=dict(default='present', choices=['present', 'absent']),
-            force=dict(type='bool', default=False),
+            force_delete_nonempty=dict(type='bool', default=False, aliases=['force']),
             tags=dict(type='dict'),
             kind=dict(type='str', default='Storage', choices=['Storage', 'StorageV2', 'BlobStorage']),
-            access_tier=dict(type='str', choices=['Hot', 'Cool'])
+            access_tier=dict(type='str', choices=['Hot', 'Cool']),
+            https_only=dict(type='bool', default=False),
+            blob_cors=dict(type='list', options=cors_rule_spec, elements='dict')
         )
 
         if HAS_AZURE:
```

Comment:
The logic in this conditional is responsible for the sanity test failures. The main issue is that this code will traceback if the Azure SDK is available but is not the correct version.

Ansible modules should not traceback when imports are missing or are outdated. Instead, the module should use fail_json to report a user-friendly error message explaining how to resolve the issue. This needs to be done after the Ansible module has been initialized.

In this particular case the module could capture the SDK version error here. After the Ansible module has been initialized it can call fail_json if an error was captured before initialization.